x-pack/solutions/observability/plugins/infra/server/lib/host_details/process_list.ts

```diff
@@ -128,35 +131,280 @@ export const getProcessList = async (
+      startTime: Date.parse(bucket.startTime.value_as_string),
+      pid: meta.process.pid,
+      state: meta.system.process.state,
+      user: meta.user.name,
+      command: bucket.key,
     };
+  });
+
+  const summary: { [p: string]: number } = result.aggregations!.summaryEvent.summary.hits.hits
+    .length
+    ? result.aggregations!.summaryEvent.summary.hits.hits[0]._source.system.process.summary
+    : {};
+
+  return {
+    processList,
+    summary,
+  };
+};
+
+const getProcessListSEMCONV = async (
+  search: ESSearchClient,
+  sourceConfiguration: InfraSourceConfiguration,
+  { hostTerm, to, sortBy, searchFilter }: Omit<ProcessListAPIRequest, 'schema'>
+) => {
+  const mandatoryFieldsFilter = MANDATORY_PROCESS_FIELDS_SEMCONV.map((field) => ({
+    exists: {
+      field,
+    },
+  }));
+
+  const filter = searchFilter ? searchFilter : [{ match_all: {} }];
+  const body = {
+    size: 0,
+    query: {
+      bool: {
+        filter: [
+          {
+            range: {
+              [TIMESTAMP_FIELD]: {
+                gte: to - 60 * 1000, // 1 minute
+                lte: to,
+                format: 'epoch_millis',
+              },
+            },
+          },
+          {
+            term: hostTerm,
+          },
+        ],
+      },
+    },
+    aggs: {
+      summaryEvent: {
+        filter: {
+          bool: {
+            must: [
+              {
+                term: {
+                  'event.dataset': 'hostmetricsreceiver.otel',
+                },
+              },
+              {
+                exists: {
+                  field: 'system.processes.count',
+                },
+              },
+            ],
+          },
+        },
+        aggs: {
+          by_status: {
+            terms: {
+              field: 'attributes.status',
+              size: 20,
+            },
+            aggs: {
+              latest_count: {
+                top_hits: {
+                  size: 1,
+                  sort: [
+                    {
+                      [TIMESTAMP_FIELD]: {
+                        order: 'desc',
+                      },
+                    },
+                  ],
+                  fields: ['system.processes.count'],
+                },
+              },
+            },
+          },
+        },
+      },
+      processes: {
+        filter: {
+          bool: {
+            must: [...filter, ...mandatoryFieldsFilter],
+          },
+        },
+        aggs: {
+          filteredProcs: {
+            terms: {
+              field: PROCESS_COMMANDLINE_FIELD,
+              size: TOP_N,
+              order: {
+                [sortBy.name]: sortBy.isAscending ? 'asc' : 'desc',
+              },
+            },
+            aggs: {
+              cpu: {
+                avg: {
+                  field: 'process.cpu.time',
+                },
+              },
+              memory: {
+                avg: {
+                  field: 'process.memory.utilization',
```

Comment:
Isn't this [`process.memory.usage`](https://opentelemetry.io/docs/specs/semconv/system/process-metrics/#metric-processmemoryusage) instead?

Reply:
same, based on tis remapper file: https://github.com/smith/opentelemetry-lib/blob/cd0e3efbcd9210a516f500896e4e1a148ea8522a/remappers/hostmetrics/process.go#L183

Reply:
`process.memory.usage` is: the amount of physical memory in use
`process.memory.utilization` is: percentage of total physical memory that is used by the process
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/hostmetricsreceiver/internal/scraper/processscraper/documentation.md#processmemoryutilization

Reply:
right! Here I believe we also need to calculate the average per state and sum them up in a bucket agg, eg: https://github.com/crespocarlos/kibana/blob/main/x-pack/solutions/observability/plugins/metrics_data_access/common/inventory_models/host/metrics/snapshot/memory.ts#L18